src/app/Recordings/ArchiveUploadModal.tsx

```diff
@@ -53,44 +55,61 @@ export const ArchiveUploadModal: React.FunctionComponent<ArchiveUploadModalProps
   const [filename, setFilename] = React.useState('' as string | undefined);
   const [uploading, setUploading] = React.useState(false);
   const [rejected, setRejected] = React.useState(false);
+  const [showCancelPrompt, setShowCancelPrompt] = React.useState(false);
+  const [abort, setAbort] = React.useState(new AbortController());
 
-  const reset = () => {
+  const reset = React.useCallback(() => {
     setUploadFile(undefined);
     setFilename('');
     setUploading(false);
     setRejected(true);
-  };
+    setShowCancelPrompt(false);
+    setAbort(new AbortController());
+  }, [setUploadFile, setFilename, setUploading, setRejected, setShowCancelPrompt, setAbort]);
```

Comment:
It's not just a performance optimization, although it can help with that. It also ensures that the lambda functions passed in as the callback are invoked with the correct current values for each of the pieces of state that are referenced - otherwise, as you say, between React renders, the callbacks may be invoked with incorrect stale state.

The dependencies arrays include other functions because those other functions are also pieces of state, which can change based on other pieces of state changing. It forms a dependency chain so that if some piece of state changes, all the possible references to it are properly updated.